alf/networks/networks.py

```diff
@@ -115,3 +118,122 @@ def __init__(self, block, input_tensor_spec=None, activation=torch.relu_):
     def forward(self, x, state=()):
         y, state = self._block(x, state)
         return self._activation(x + y), state
+
+
+class TemporalStack(Network):
+    """Stack features temporally.
+
+    Suppose input_size=(), stack_size=2, pooling_size=2, the following table
+    shows the output of different mode for an input sequence of 1,2,3,4,5 (ignoring
+    batch dimension)
+
+           1,        2,        3,        4,          5
+    skip:  [0, 1],   [0, 1],   [1, 3],   [1, 3],     [3, 5]
+    avg:   [0, 0],   [0, 1.5], [0, 1.5], [1.5, 3.5], [1.5, 3.5]
+    max:   [0, 0],   [0, 2],   [0, 2],   [2, 4],     [2, 4]
+
+    Example:
+
+    .. code-block:: python
+
+        # A temporal CNN with progressively large temporal receptive field.
+        cnn = alf.networks.Sequential([
+            alf.networks.TemporalStack(256, 3, 1),
+            torch.nn.Flatten(),
+            alf.layers.FC(768, 256, activation=torch.relu_)
+            alf.networks.TemporalStack(256, 3, 2),
+            torch.nn.Flatten(),
+            alf.layers.FC(768, 256, activation=torch.relu_)
+            alf.networks.TemporalStack(256, 3, 4),
+            torch.nn.Flatten(),
+            alf.layers.FC(768, 256, activation=torch.relu_)])
+
+    """
+
+    def __init__(self,
+                 input_size,
+                 stack_size,
+                 pooling_size=1,
+                 dtype=torch.float32,
+                 mode='skip'):
+        """
+        Args:
+            input_size (int|tuple[int]): shape of the input
```

Comment:
It might be useful to comment on which dim is assumed to be the temporal dimension, especially when input_size is tuple (multi-dimensional input).